src/include/txn_inline.h

```diff
@@ -917,8 +918,14 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
      *    WT_UPDATE.start_ts = WT_TS_NONE;
      *    WT_UPDATE.prepare_state = WT_PREPARE_INIT;
      *    WT_UPDATE.flags = 0;
+     *
+     * If there is no value ensure that the memory allocated is never less than the size
+     * reported by sizeof(). Otherwise bit-exact tools like MSan may infer the structure is not
+     * completely initialized.
      */
-    WT_RET(__wt_calloc(session, 1, WT_UPDATE_SIZE + (value == NULL ? 0 : value->size), &upd));
+    allocsz =
+      WT_MAX((WT_UPDATE_SIZE + ((value != NULL) ? value->size : 0)), WT_UPDATE_SIZE_NOVALUE);
```

Comment:
This suggestion is borderline a style preference, but I don't like ternary statements along with other conditionals in a single line. I know this code already looked similar, and I'm sure there are plenty of examples of that in WiredTiger, which is why it's more of a personal preference. That long caveat was meant to mean this is a point of conversation rather than a definite ask...

I find:

```
+    if (value == NULL || value->size == 0)
+           allocsz = WT_UPDATE_SIZE_NOVALUE;
+    else
+           allocsz = WT_UPDATE_SIZE + value->size;
```

Easier to read and understand than the line you are proposing here.

Reply:
The ternary in this context does not help readability to be sure. It was included in order to retain the original semantics of the function which appears to imply that value->size may not be zero even if value->data == NULL.

The conditional code is brittle in that if amount of natural padding in WT_UPDATE increases in the future, so that:
```
assert(value->size > 0);
assert(WT_UPDATE_SIZE + value->size < WT_UPDATE_SIZE_NOVALUE);
```
This is not true when using `WT_MAX` would cover this circumstance. However that invalidates the name `WT_UPDATE_SIZE_NOVALUE` which might be called `WT_UDPATE_SIZE_NATURAL`.
The determination as to whether the value size is zero could be moved to a separate line removing the need to a ternary.

Reply:
Refactored consistent with our conversation. 🤞